File: packages/core/frontend/src/types/resources/endpoint.ts
import type { ManagedField } from "./shared";

export interface EndpointAddress {
  IP: string;
  Hostname: string;
  TargetName: string;
}

export interface EndpointPort {
  Name: string;
  Port: number;
  Protocol: string;
}

export interface EndpointSubset {
  Addresses: EndpointAddress[];
  Ports: EndpointPort[];
}

export interface Endpoint {
  Name: string;
  Namespace: string;
  Endpoints: string;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Subsets: EndpointSubset[];
}

File: packages/core/kube/dto/endpoint.go
package dto

type EndpointRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type EndpointAddress struct {
	IP         string
	Hostname   string
	TargetName string
}

type EndpointPort struct {
	Name     string
	Port     int32
	Protocol string
}

type EndpointSubset struct {
	Addresses []EndpointAddress
	Ports     []EndpointPort
}

type Endpoint struct {
	Name      string
	Namespace string
	Endpoints string
	Age       string

	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	Subsets       []EndpointSubset
}
